cime_config/acme/machines/template.case.run

```diff
@@ -1,416 +1,327 @@
-
-    $logger->info( " BUILDNML SCRIPT HAS FINISHED SUCCESSFULLY");
-    $logger->info( "-------------------------------------------------------------------------");
-    $logger->info( " PRESTAGE SCRIPT STARTING");
-    $logger->info(" - Case input data directory, DIN_LOC_ROOT, is $config{'DIN_LOC_ROOT'}");
-    $logger->info( " - Checking the existence of input datasets in DIN_LOC_ROOT");
-
-}
-
-# -------------------------------------------------------------------------
-# Run the model system
-# -------------------------------------------------------------------------
-sub runModel()
-{
-    system("sleep 10");
-    my $localtime = localtime;
-    # Set OMP_NUM_THREADS
-    $ENV{'OMP_NUM_THREADS'} = {{ thread_count }};
-
-    $logger->info("$localtime MODEL EXECUTION BEGINS HERE");
-    # Run the model
-    my $cmd = "{{ mpirun }}";
-    $cmd = SetupTools::expand_xml_var($cmd, \%config);
 
-    chdir $config{'RUNDIR'};
-    $logger->info("Running '$cmd'\n");
-    system($cmd);
+###############################################################################
+def parse_command_line(args):
+###############################################################################
+
+    CIME.utils.get_model()
+
+    parser = argparse.ArgumentParser()
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    args = parser.parse_args()
+
+    CIME.utils.handle_standard_logging_options(args)
+
+###############################################################################
+def preRunCheck(case):
+###############################################################################
+
+    # Pre run initialization code..
+    model          = CIME.utils.get_model()
+    caseroot       = case.get_value("CASEROOT")
+    cimeroot       = case.get_value("CIMEROOT")
+    din_loc_root   = case.get_value("DIN_LOC_ROOT")
+    compiler       = case.get_value("COMPILER")
+    debug          = case.get_value("DEBUG")
+    mach           = case.get_value("MACH")
+    lbquery        = case.get_value("LBUQERY")
+    batchsubmit    = case.get_value("BATCHSUBMIT")
+    mpilib         = case.get_value("MPILIB")
+    rundir         = case.get_value("RUNDIR")
+    build_complete = case.get_value("BUILD_COMPLETE")
+
+    # check for locked files.
+    check_lockedfiles()
+    logger.debug("check_lockedfiles OK")
+
+    # check that build is done
+    expect (build_complete,
+            "BUILD_COMPLETE is not true\nPlease rebuild the model interactively")
+    logger.debug("build complete is %s " %build_complete)
+
+    # load the module environment...
+    env_module = EnvModule(mach, compiler, cimeroot, caseroot, mpilib, debug)
+    env_module.load_env_for_case()
+
+    # set environment variables
+    # currently need to set mpilib as an environment variable- otherwise
+    # the model dies at runtime 
+    # FIXME: is there a way to remove this requirement?
+    if mpilib == "mpi-serial":
+        del os.environ["MP_MPILIB"]
+    else:
+        os.environ["MPILIB"] = mpilib
```

Comment:
Please put a blank line to visually separate the previous if/else block from the next one.